revenue_app/presto_connection.py

```diff
@@ -0,0 +1,69 @@
+import pandas as pd
+from pyhive import presto
+from pyhive.exc import DatabaseError
+
+
+class PrestoError(Exception):
+    pass
+
+
+def read_sql(file_name):
+    with open('sql/{}.sql'.format(file_name), 'r') as fd:
+        sql_file = fd.read()
+    return sql_file
+
+
+def query_presto(start_date, end_date, okta_username, okta_password, query, query_name):
+    try:
+        connection = presto.connect(
+            'presto-tableau.prod.dataf.eb',
+            8443,
+            okta_username,
+            password=okta_password,
+            protocol='https',
+        )
+        dataframe = pd.read_sql(query.format(start_date, end_date), connection)
+        connection.close()
+    except Exception as exception:
+        return handle_presto_exception(exception, query_name)
+    return dataframe
+
+
+def make_query(start_date, end_date, okta_username, okta_password, query_name):
+    query = read_sql(query_name)
+    dataframe = query_presto(start_date, end_date, okta_username, okta_password, query, query_name)
+    return dataframe
+
+
+def handle_presto_exception(exception, query_name=None):
+    if isinstance(exception, pd.io.sql.DatabaseError):
+        error = exception.args[0].split('\n\n')[1]
+        if 'NewConnectionError' in error:
+            # Raised when not connected to the VPN
+            message = error.split('>:')[1].strip()
+            message += '.<br>Check you are connected to the VPN.'
+        elif 'SSLCertValidationError' in error:
+            # Raised when certificate not in env var
+            message = error.split("(1, '")[1].split("')))")[0].strip()
+            message += '.<br>Update your certificate '
+            message += '(<a href="https://docs.evbhome.com/intro/self_signed_certs.html">link</a>).'
+        elif 'Invalid credentials' in error:
+            # Raised when wrong okta username/password
+            message = error.split('<pre>')[1].split('</pre>')[0].strip()
+            message += '.<br>Check your okta username and password.'
+        else:
+            message = 'Unknown error.<br>'
+            message += error
```

Comment:
I know this are all exceptions coming from Presto but they really feel different exceptions.